invoices/invoiceregistry.go

```diff
@@ -1043,3 +955,114 @@ func (i *InvoiceRegistry) HodlUnsubscribeAll(subscriber chan<- interface{}) {
 
 	delete(i.hodlReverseSubscriptions, subscriber)
 }
+
+// invoiceUpdater is an object that is responsible for updating the invoice
+// state and storing actions that need to be carried out after the invoice is
+// updated.
+type invoiceUpdater struct {
+	// update input fields
```

Comment:
wip?

Reply:
no, not wip. it is a 'group comment'. I don't want to describe all these obvious fields again.

Reply:
Perhaps move the group comment to the comment on the struct then? It sticks out a bit relative to our regular comment style.

Reply:
Changed this part to a `invoiceUpdateCtx` and pass back the outcome as return values.